modules/custom-status/lib/components/custom-status-sidebar.js

```diff
@@ -0,0 +1,84 @@
+import { Button, SelectControl } from '@wordpress/components';
+import { dispatch, select, subscribe } from '@wordpress/data';
+import { PluginPostStatusInfo } from '@wordpress/edit-post';
+import { store as editorStore } from '@wordpress/editor';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+import { isUsingWorkflowStatus } from '../util';
+
+/**
+ * Map Custom Statuses as options for SelectControl
+ */
+const statusOptions = VW_CUSTOM_STATUSES.status_terms.map( customStatus => ( {
+	label: customStatus.name,
+	value: customStatus.slug,
+} ) );
+
+/**
+ * Subscribe to changes so we can set a default status.
+ */
+subscribe( function () {
+	const postId = select( editorStore ).getCurrentPostId();
+	if ( ! postId ) {
+		// Post isn't ready yet so don't do anything.
+		return;
+	}
+
+	// For new posts, we need to force the default custom status which is the first entry.
+	const isCleanNewPost = select( editorStore ).isCleanNewPost();
+
+	if ( isCleanNewPost ) {
+		dispatch( editorStore ).editPost( {
+			status: statusOptions[ 0 ].value,
+		} );
+	}
+} );
+
+/**
+ * Custom status component
+ * @param object props
+ */
+export default function CustomStatusSidebar( { onUpdateStatus, postType, status } ) {
+	const [ isEditingStatus, setIsEditingStatus ] = useState( false );
+
+	const handleChangeStatus = statusSlug => {
+		onUpdateStatus( statusSlug );
+		dispatch( editorStore ).savePost();
+		setIsEditingStatus( false );
+	};
+
+	if ( ! isUsingWorkflowStatus( postType, status ) ) {
+		return null;
+	}
+
+	const customStatusName = VW_CUSTOM_STATUSES.status_terms.find( t => t.slug === status )?.name;
```

Comment:
Minor: Would replace the t with term. Not a fan of one word variables unless it's a counter

Reply:
Fixed in https://github.com/Automattic/vip-workflow-plugin/pull/24/commits/240034cef39a73fa014a1ade855fa48702368294. I usually also prefer longer variable names, but for one-liner lambdas I sometimes take the lazy route. Happy to stick with longer, though.